editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -2941,16 +2944,235 @@ void Node3DEditorViewport::_notification(int p_what) {
+							// YZ plane
+							corner_point = Vector3(start_pos.x, start_pos.y, end_pos.z);
+						} else if (Math::abs(delta.x) > 0.0 && Math::abs(delta.y) > 0.0 && Math::abs(delta.z) > 0.0) {
+							// All three axes
+							corner_point = Vector3(end_pos.x, start_pos.y, start_pos.z);
+						} else {
+							corner_point = end_pos;
+						}
+
+						triangle_mesh->clear_surfaces();
+						triangle_mesh->surface_begin(Mesh::PRIMITIVE_LINES);
+
+						Vector3 triangle_camera_dir = (camera->get_transform().origin - center).normalized();
+						real_t triangle_offset_distance = 0.01;
+
+						triangle_mesh->surface_add_vertex(start_pos + triangle_camera_dir * triangle_offset_distance);
+						triangle_mesh->surface_add_vertex(corner_point + triangle_camera_dir * triangle_offset_distance);
+						triangle_mesh->surface_add_vertex(corner_point + triangle_camera_dir * triangle_offset_distance);
+						triangle_mesh->surface_add_vertex(end_pos + triangle_camera_dir * triangle_offset_distance);
+
+						triangle_mesh->surface_end();
+
+						triangle_mesh_xray->clear_surfaces();
+						triangle_mesh_xray->surface_begin(Mesh::PRIMITIVE_LINES);
+
+						triangle_mesh_xray->surface_add_vertex(start_pos);
+						triangle_mesh_xray->surface_add_vertex(corner_point);
+						triangle_mesh_xray->surface_add_vertex(corner_point);
+						triangle_mesh_xray->surface_add_vertex(end_pos);
+
+						triangle_mesh_xray->surface_end();
+					}
+				} else {
+					ImmediateMesh *triangle_mesh = Object::cast_to<ImmediateMesh>(ruler_triangle_lines->get_mesh().ptr());
+					ImmediateMesh *triangle_mesh_xray = Object::cast_to<ImmediateMesh>(ruler_triangle_lines_xray->get_mesh().ptr());
+
+					if (triangle_mesh) {
+						triangle_mesh->clear_surfaces();
+					}
+					if (triangle_mesh_xray) {
+						triangle_mesh_xray->clear_surfaces();
+					}
+				}
+
+				if (show_components) {
+					Vector3 delta = end_pos - start_pos;
+
+					const real_t threshold = 0.001;
+					if (Math::abs(delta.x) < threshold) {
+						delta.x = 0.0;
+					}
+					if (Math::abs(delta.y) < threshold) {
+						delta.y = 0.0;
+					}
+					if (Math::abs(delta.z) < threshold) {
+						delta.z = 0.0;
+					}
+
+					int active_axes = 0;
+					if (Math::abs(delta.x) > 0.0) {
+						active_axes++;
+					}
+					if (Math::abs(delta.y) > 0.0) {
+						active_axes++;
+					}
+					if (Math::abs(delta.z) > 0.0) {
+						active_axes++;
+					}
+
+					String x_text = Math::abs(delta.x) > 0.0 ? TS->format_number(vformat("X: %.3f m", Math::abs(delta.x))) : "";
+					String y_text = Math::abs(delta.y) > 0.0 ? TS->format_number(vformat("Y: %.3f m", Math::abs(delta.y))) : "";
+					String z_text = Math::abs(delta.z) > 0.0 ? TS->format_number(vformat("Z: %.3f m", Math::abs(delta.z))) : "";
+
+					Vector3 corner_point;
+
+					if (active_axes >= 2) {
+						if (Math::abs(delta.x) > 0.0 && Math::abs(delta.y) > 0.0 && Math::abs(delta.z) == 0.0) {
```

Comment:
```suggestion
						if (delta.z == 0.0) {
```
For the if-else chain here, any coordinate being negligible suggests that the other two are both active considering the block it's in, so I think this (and the changes I proposed for 3091 and 3106) can help simplify these checks without altering their functionality in any way. The Math::abs check was removed here (and on those other two checks) for similar reasoning to the suggestion for lines 2993-3007.